plaidml/bridge/openvino/ops/prior_box.cpp

```diff
@@ -0,0 +1,383 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;  // NOLINT[build/namespaces]
+using ngraph::opset4::PriorBox;
+
+namespace PlaidMLPlugin {
+
+void registerPriorBox() {
+  registerOp("PriorBox", [](const Context& ctx) {
+    auto* layer = ngraph::as_type<PriorBox>(ctx.layer);
+    IE_ASSERT(ctx.operands.size() == 2);
+    auto* input_shape_ngraph_op = ngraph::as_type<ngraph::op::Constant>(ctx.layer->get_input_node_ptr(0));
+    std::vector<int> input_shape = input_shape_ngraph_op->get_vector<int>();
+    int H = input_shape[0];
+    int W = input_shape[1];
+    auto* image_shape_ngraph_op = ngraph::as_type<ngraph::op::Constant>(ctx.layer->get_input_node_ptr(1));
+    std::vector<int> image_shape = image_shape_ngraph_op->get_vector<int>();
+    int IH = image_shape[0];
+    int IW = image_shape[1];
+
+    const ngraph::op::PriorBoxAttrs& attrs = layer->get_attrs();
+
+    int num_of_prior = ngraph::op::PriorBox::number_of_priors(attrs);
+
+    std::vector<float> aspect_ratios = {1.0f};
+    for (const auto& aspect_ratio : attrs.aspect_ratio) {
+      bool exist = false;
+      for (const auto existed_value : aspect_ratios) exist |= std::fabs(aspect_ratio - existed_value) < 1e-6;
+
+      if (!exist) {
+        aspect_ratios.push_back(aspect_ratio);
+        if (attrs.flip) {
+          aspect_ratios.push_back(1.0f / aspect_ratio);
+        }
+      }
+    }
+    std::vector<float> aspect_ratios_scale_size;
+    for (float ar : aspect_ratios) {
+      if (std::fabs(ar - 1.0f) < 1e-6) {
+        continue;
+      }
+      aspect_ratios_scale_size.push_back(std::sqrt(ar));
+    }
+
+    std::vector<float> variance = attrs.variance;
+    IE_ASSERT(variance.size() == 1 || variance.size() == 4 || variance.empty());
+    if (variance.empty()) variance.push_back(0.1f);
+
+    float step = attrs.step, step_x, step_y;
+    auto min_size = attrs.min_size;
+    if (!attrs.scale_all_sizes) {
+      // mxnet-like PriorBox
+      if (step == -1)
+        step = 1.f * IH / H;
+      else
+        step *= IH;
+      for (auto& size : min_size) size *= IH;
+    }
+    if (step == 0) {
+      step_x = static_cast<float>(IW) / W;
+      step_y = static_cast<float>(IH) / H;
+    } else {
+      step_x = step;
+      step_y = step;
+    }
```

Comment:
use something like 
step=(step==-1)?1.f * IH / H:IH
step_x=(step == 0)?static_cast<float>(IW) / W:step;
step_y=(step == 0)?static_cast<float>(IH) / H:step;

Reply:
when step==-1, the second operator is *=, so better not change style,  i will change "step == 0" path

Reply:
ok, I see, thanks.